src/app/qgscrashreport.h

```diff
@@ -0,0 +1,114 @@
+/***************************************************************************
+  qgscrashreport.h - QgsCrashReport
+
+ ---------------------
+ begin                : 16.4.2017
+ copyright            : (C) 2017 by Nathan Woodrow
+ email                : woodrow.nathan@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSCRASHREPORT_H
+#define QGSCRASHREPORT_H
+
+#include "qgis_app.h"
+
+#include <QObject>
+
+
+/**
+ * Include information to generate user friendly crash report for QGIS.
+ */
+class APP_EXPORT QgsCrashReport
+{
+  public:
+
+    /**
+     * Represents a line from a stack trace.
+     */
+    struct StackLine
+    {
+      QString ModuleName;
+      QString SymbolName;
+      QString FileName;
+      QString LineNumber;
+
+      /**
+       * Check if this stack line is part of QGIS.
+       * @return True if part of QGIS.
+       */
+      bool IsQGISModule();
+
+      /**
+       * Check if this stack line is valid.  Considered valid when the filename and line
+       * number are known.
+       * @return True of the line is valid.
+       */
+      bool IsValid();
+    };
+
+    /**
+     * Include information to generate user friendly crash report for QGIS.
+     */
+    QgsCrashReport();
+
+  public:
+    enum Flag
+    {
+      Stack                = 1,
```

Comment:
These numbers aren't right - they need to be unique bits. Use

    Stack = 1 << 0,
    Plugins = 1 << 1,
    ProjectDetails = 1 << 2,
    SystemInfo = 1 << 3 (8, not 6)
etc